### PR TITLE
PathManager enhancements

### PR DESCRIPTION
**Pull request by Benglin**
Changes
-----
1. Constant strings that might be of interest to external code are exposed by `PathManager`
2. `IPathManager` now exposes new properties: `UserDataDirectory` and `CommonDataDirectory`

@aparajit-pratap PTAL

### CODE REVIEW

File: src/DynamoCore/Core/PathManager.cs [hunk 1]
```diff
@@ -139,23 +190,32 @@ internal PathManager(string corePath, IPathResolver pathResolver)
...
+            userDataDir = GetUserDataFolder();
 
-            userDefinitions = CreateFolder(Path.Combine(userDataDir, "definitions"));
-            logDirectory = CreateFolder(Path.Combine(userDataDir, "Logs"));
-            packagesDirectory = CreateFolder(Path.Combine(userDataDir, "packages"));
-            preferenceFilePath = Path.Combine(userDataDir, "DynamoSettings.xml");
+            userDefinitions = Path.Combine(userDataDir, DefinitionsDirectoryName);
+            logDirectory = Path.Combine(userDataDir, LogsDirectoryName);
+            packagesDirectory = Path.Combine(userDataDir, PackagesDirectoryName);
+            preferenceFilePath = Path.Combine(userDataDir, PreferenceSettingsFileName);
```
Comment (Benglin) [inline src/DynamoCore/Core/PathManager.cs]: Actual folder creations are now moved into `EnsureDirectoryExistence`.

File: src/DynamoCore/Core/PathManager.cs [hunk 2]
```diff
@@ -131,7 +181,8 @@ internal PathManager(string corePath, IPathResolver pathResolver)
             }
 
             dynamoCoreDir = corePath;
-            if (!File.Exists(Path.Combine(dynamoCoreDir, "DynamoCore.dll")))
+            var assemblyPath = Path.Combine(dynamoCoreDir, "DynamoCore.dll");
+            if (!File.Exists(Path.Combine(assemblyPath)))
```
Comment (aparajit-pratap) [inline src/DynamoCore/Core/PathManager.cs]: you mean `if(!File.Exists(assemblyPath))`?

**Comment by aparajit-pratap** [comment]
@Benglin Aside from the comment above, LGTM. Thanks for this!

File: src/DynamoCore/Core/PathManager.cs (hunk 2 above)
Comment (Benglin) [inline src/DynamoCore/Core/PathManager.cs]: Isn't... that the same?

Comment (Benglin) [inline src/DynamoCore/Core/PathManager.cs]: AH! Good catch, thanks!

**Comment by Benglin** [comment]
Thanks @aparajit-pratap!

**Comment by Benglin** [comment]
Yes, the base of `UserDataDirectory` is obtained from `IPathResolver`, which is external. If you give `PathManager` constructor an instance of that `IPathResolver`, it will get the base folder path from there, and then form the rest on its own. For example, the `IPathResolver` will return base path as `C:\ProgramData\Dynamo`, then the `PathManager` will then turn that into `C:\ProgramData\Dynamo\0.8`. Does that make sense?

**Pull request merged**